internal/form/submit/service_test.go

```diff
@@ -0,0 +1,257 @@
+package submit_test
+
+import (
+	"NYCU-SDC/core-system-backend/internal/form/submit"
+	"context"
+	"errors"
+	"testing"
+
+	"NYCU-SDC/core-system-backend/internal/form/question"
+	"NYCU-SDC/core-system-backend/internal/form/response"
+	"NYCU-SDC/core-system-backend/internal/form/shared"
+
+	"NYCU-SDC/core-system-backend/internal/form/submit/mocks"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+type fakeAnswerable struct {
+	q           question.Question
+	validateErr error
+}
+
+func (f fakeAnswerable) Question() question.Question {
+	return f.q
+}
+func (f fakeAnswerable) Validate(value string) error {
+	return f.validateErr
+}
+
+func ans(qID uuid.UUID, val string) shared.AnswerParam {
+	return shared.AnswerParam{QuestionID: qID.String(), Value: val}
+}
+
+func makeQ(questionID uuid.UUID, t question.QuestionType, validateErr error) question.Answerable {
+	return fakeAnswerable{
+		q: question.Question{
+			ID:   questionID,
+			Type: t,
+		},
+		validateErr: validateErr,
+	}
+}
+
+func TestSubmitService_Submit(t *testing.T) {
+	type Params struct {
+		formID  uuid.UUID
+		userID  uuid.UUID
+		answers []shared.AnswerParam
+		qs      []question.Answerable
```

Comment:
Consider to rename it to `questions` for better readability
```suggestion
		questions      []question.Answerable
```